projects/plugins/jetpack/_inc/lib/class-jetpack-ai-helper.php

```diff
@@ -154,14 +154,28 @@ public static function get_gpt_completion( $content, $post_id ) {
 				\require_lib( 'openai' );
 			}
 
-			$result = ( new OpenAI( 'openai', array( 'post_id' => $post_id ) ) )->request_gpt_completion( $content );
+			// Set the content for chatGPT endpoint
+			$data = array(
+				array(
+					'role'    => 'user',
+					'content' => $content,
+				),
+			);
+
+			$result = ( new OpenAI( 'openai', array( 'post_id' => $post_id ) ) )->request_chat_completion(
+				$data
+			);
+
 			if ( is_wp_error( $result ) ) {
 				return $result;
 			}
+
+			$response = $result->choices[0]->message->content;
+
 			// In case of Jetpack we are setting a transient on the WPCOM and not the remote site. I think the 'get_current_user_id' may default for the connection owner at this point but we'll deal with this later.
-			set_transient( self::transient_name_for_completion(), $result, self::$text_completion_cooldown_seconds );
+			set_transient( self::transient_name_for_completion(), $response, self::$text_completion_cooldown_seconds );
```

Comment:
What do you think about lowering cooldown to 15s?

Reply:
Yes, it makes sense. At that stage this is merely a burst protection than anything else... It can totally confuse the user if too long...